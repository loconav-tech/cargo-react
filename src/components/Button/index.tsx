import * as React from 'react';
import styled, { css } from 'styled-components';

interface ButtonProps {
  children?: React.ReactChild;
  className?: string;
  radius?: number;
  size?: "small" | "medium" | "big";
  look?: "primary" | "secondary" | "dark" | "light";
}
const Button: React.StatelessComponent<ButtonProps> = p => (
  <button className={p.className}>{p.children}</button>
);

const sizes = {
  small: css`
    padding: 5px 20px;
    font-size: 12px;
  `,
  medium: css`
    padding: 10px 30px;
    font-size: 14px;
  `,
  big: css`
    padding: 15px 40px;
    font-size: 18px;
  `
};

const looks = {
  primary: css`
    border: 1px solid tomato;
    background: tomato;
    color: white;
  `,
  secondary: css`
    border: 1px solid palevioletred;
    background: palevioletred;
    color: white;
  `,
  dark: css`
    border: 1px solid #273444;
    background: #273444;
    color: white;
  `,
  light: css`
    border: 1px solid #eff2f7;
    background: #f9fafc;
    color: #273444;
  `
};

const StyledButton = styled(Button)`
  ${({ size = "small" }) => sizes[size]};
  ${({ look = "primary" }) => looks[look]};

  outline: none;
  border-radius: ${({ radius = 5 }) => radius}px;
`;

export default StyledButton;
