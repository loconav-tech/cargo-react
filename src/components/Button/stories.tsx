import { storiesOf } from '@storybook/react';
import * as React from 'react';
import Button from "./index";

storiesOf("Button", module)
  .add("General", () => (
    <Button>Click Me</Button>
  ))
  .add("Secondary Medium", () => (
    <Button look="secondary" size="medium">
      Click Me
    </Button>
	))
	.add("Dark Medium", () => (
    <Button look="dark" size="medium">
      Click Me
    </Button>
	))
	.add("Light Big With Radius", () => (
    <Button look="light" size="big" radius={100}>
      Click Me
    </Button>
	));
