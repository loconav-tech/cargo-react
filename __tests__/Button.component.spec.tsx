import * as React from "react";
import Button from "../src/components/Button/index";
import * as renderer from "react-test-renderer";

test("Component should match with general component", () => {
  const component = renderer.create(<Button>Click Me</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("Component should match with Secondary Medium", () => {
  const component = renderer.create(
		<Button look="secondary" size="medium">
		Click Me
		</Button>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("Component should match with Dark Medium", () => {
  const component = renderer.create(
		<Button look="dark" size="medium">
		Click Me
		</Button>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Component should match with Look light size Big and Radius", () => {
  const component = renderer.create(
		<Button look="light" size="big" radius={100}>
		Click Me
		</Button>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
