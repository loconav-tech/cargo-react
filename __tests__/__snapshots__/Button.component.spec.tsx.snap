// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should match with Dark Medium 1`] = `
<button
  className="sc-bdVaJa jgfyHn"
>
  Click Me
</button>
`;

exports[`Component should match with Look light size Big and Radius 1`] = `
<button
  className="sc-bdVaJa hmFDPf"
>
  Click Me
</button>
`;

exports[`Component should match with Secondary Medium 1`] = `
<button
  className="sc-bdVaJa fgQjOJ"
>
  Click Me
</button>
`;

exports[`Component should match with general component 1`] = `
<button
  className="sc-bdVaJa bojbCQ"
>
  Click Me
</button>
`;
